tests/test_magnetostatics.py

```diff
@@ -0,0 +1,183 @@
+    calculator = CalculatorDipole(
+        potential=PotentialDipole(smearing=smearing, dtype=torch.float64),
+        full_neighbor_list=False,
+        lr_wavelength=0.1,
+        dtype=torch.float64,
+    )
+    pot = calculator(
+        dipoles=system.dipoles,
+        cell=system.cell,
+        positions=system.positions,
+        neighbor_indices=system.neighbor_indices,
+        neighbor_vectors=system.neighbor_vectors,
+    )
+    result = torch.einsum("ij,ij->i", pot, system.dipoles).sum()
+    # result is calculated using espressomd DipolarP3M with the same parameters and mesh
+    # size 64
+    expected_result = torch.tensor(-0.30848574939287954, dtype=torch.float64)
+    assert torch.isclose(result, expected_result, atol=1e-4), (
+        f"Expected {expected_result}, but got {result}"
+    )
+
+
+frames = read(DIPOLES_TEST_FRAMES, ":3")
+cutoffs = [3.9986718930, 4.0000000000, 4.7363281250]
+alphas = [0.8819831493, 0.8956299559, 0.7215211182]
+energies = [frame.get_potential_energy() for frame in frames]
+forces = [frame.get_forces() for frame in frames]
+
+
+@pytest.mark.parametrize(
+    ("frame", "cutoff", "alpha", "energy", "force"),
+    zip(frames, cutoffs, alphas, energies, forces),
+)
+def test_magnetostatic_ewald_crystal(frame, cutoff, alpha, energy, force):
+    smearing = (1 / (2 * alpha**2)) ** 0.5
+    calc = CalculatorDipole(
+        potential=PotentialDipole(smearing=smearing, dtype=torch.float64),
+        full_neighbor_list=False,
+        lr_wavelength=0.1,
+        dtype=torch.float64,
+        prefactor=eV_A,
+    )
+    positions = torch.tensor(frame.get_positions(), dtype=torch.float64)
+    dipoles = torch.tensor(frame.get_array("dipoles"), dtype=torch.float64)
+    cell = torch.tensor(frame.get_cell().array, dtype=torch.float64)
+    neighbor_indices, neighbor_shifts = neighbor_list(
+        positions=positions,
+        periodic=True,
+        box=cell,
+        cutoff=cutoff,
+        full_neighbor_list=False,
+        neighbor_shifts=True,
+    )
+    positions.requires_grad = True
+    neighbor_distance_vectors = compute_distance_vectors(
+        positions=positions,
+        neighbor_indices=neighbor_indices,
+        cell=cell,
+        neighbor_shifts=neighbor_shifts,
+    )
+    pot = calc(
+        dipoles=dipoles,
+        cell=cell,
+        positions=positions,
+        neighbor_indices=neighbor_indices,
+        neighbor_vectors=neighbor_distance_vectors,
+    )
+
+    result = torch.einsum("ij,ij->", pot, dipoles)
+    expected_result = torch.tensor(energy, dtype=torch.float64)
+    assert torch.isclose(result, expected_result, atol=1e-4), (
+        f"Expected {expected_result}, but got {result}"
+    )
+
+    forces = -torch.autograd.grad(result, positions)[0]
+    expected_forces = torch.tensor(force, dtype=torch.float64)
+    assert torch.allclose(forces, expected_forces, atol=1e-4), (
+        f"Expected {expected_forces}, but got {forces}"
+    )
```

Comment:
Please also test that the gradient of the cell is not nan. We have no reference values (yet) but at least with this we can test that the backprop runs.

We can use lammps to compute the stress.